source/abstract/memory/ExpHeap.hh

```diff
@@ -57,6 +57,9 @@ public:
         } fields;
     } m_attribute;
     u32 m_size;
+#ifdef BUILD_DEBUG
+    u32 m_tag;
```

Comment:
I just remember you saying something about intentionally declaring this such that there is implicit padding. Is it worth leaving a comment here to elaborate on why you wanted to do that?

Reply:
Is it necessary? If we ever need to expand this further, we have a place to put new small variables without increasing the overhead of the block.

We do not reach more than 400 allocations under normal circumstances from what I've seen - I'm considering limiting this to a u16.